internal/appsec/dyngo/instrumentation/httpsec/actions.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package httpsec
+
+import (
+	"net/http"
+)
+
+// Action is used to identify any action kind
+type Action interface {
+	isAction()
+}
+
+// BlockRequestAction is the parameter struct used to perform actions of kind ActionBlockRequest
+type BlockRequestAction struct {
+	Action
+	// Status is the return code to use when blocking the request
+	Status int
+	// Template is the payload template to use to write the response (html or json)
+	Template string
+	// handler is the http handler be used to block the request (see wrap())
+	handler http.Handler
+}
+
+func (*BlockRequestAction) isAction() {}
+
+func blockedPayload(a *BlockRequestAction) []byte {
+	payload := BlockedTemplateJSON
+	if a.Template == "html" {
+		payload = BlockedTemplateHTML
+	}
+	return payload
+}
+
+// ActionsHandler handles actions registration and their application to operations
+type ActionsHandler struct {
+	actions map[string]Action
+}
+
+// NewActionsHandler returns an action handler holding the default ASM actions.
+// Currently, only the default "block" action is supported
+func NewActionsHandler() *ActionsHandler {
+	handler := ActionsHandler{
+		actions: map[string]Action{},
+	}
+	// Register the default "block" action as specified in the RFC for HTTP blocking
+	handler.RegisterAction("block", &BlockRequestAction{
+		Status:   403,
+		Template: "html",
+	})
```

Comment:
The spec says the default blocking action is:

> The default block action is defined as follow:
id: block
type: block_request
parameters:
   status_code: 403
   type: auto
   grpc_status_code: 10

You don't seem to follow it here.